core/src/lightclient/mod.rs

```diff
@@ -82,4 +122,61 @@ impl StarknetClient {
         let block = starknet_block_to_eth_block(starknet_block);
         Ok(block)
     }
+
+    /// Get the number of transactions in a block given a block id.
+    /// The number of transactions in a block.
+    ///
+    /// # Arguments
+    ///
+    ///
+    ///
+    /// # Returns
+    ///
+    ///  * `block_number(u64)` - The block number.
+    ///
+    /// `Ok(Bytes)` if the operation was successful.
+    /// `Err(LightClientError)` if the operation failed.
+    async fn get_code(
+        &self,
+        ethereum_address: Address,
+        starknet_block_id: StarknetBlockId,
+    ) -> Result<Bytes, LightClientError> {
+        // Convert the Ethereum address to a hex-encoded string
+        let address_hex = hex::encode(ethereum_address);
+        // Convert the hex-encoded string to a FieldElement
+        let ethereum_address_field_element = FieldElement::from_hex_be(&address_hex).unwrap();
+
+        // Prepare the calldata for the get_starknet_contract_address function call
+        let tx_calldata_vec = vec![ethereum_address_field_element];
+        let request = FunctionCall {
+            contract_address: self.kakarot_account_registry,
+            entry_point_selector: selector!("get_starknet_contract_address"),
+            calldata: tx_calldata_vec,
+        };
+        // Make the function call to get the Starknet contract address
+        let starknet_contract_address = self.client.call(request, &starknet_block_id).await?;
+        // Concatenate the result of the function call
+        let concatenated_result = starknet_contract_address
```

Comment:
if this is going to be reused at every .call of the client, let's make this into a named util:)! 
I understand that the code of the contract is given in an array of 16bytes chunks that we need to concatenate, right?
let's make a pure function util and test it